ipa-core/src/protocol/dp/distributions.rs

```diff
@@ -125,4 +216,191 @@ mod tests {
         };
         check(&nd, &mut rng, 1_u8);
     }
+
+    /// Tests for Geometric
+    #[test]
+    fn test_geometric_constructor() {
+        // should fail for negative probability
+        let p = -1.0;
+        let expected = Err(Error::BadGeometricProb(p));
+        let actual = Geometric::new(p);
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn test_geometric_sample_dist() {
+        let mut rng = rand::thread_rng();
+        let p = 0.5; // success probability
+        let geometric = Geometric::new(p).expect("Geometric not constructed");
+        let mut histogram = HashMap::new();
+        let num_samples = 100_000;
+        for _ in 0..num_samples {
+            let sample = geometric.sample(&mut rng);
+            *histogram.entry(sample).or_insert(0) += 1;
+        }
+        #[allow(clippy::cast_precision_loss)]
+        for x in 0..100 {
+            let observed_probability = histogram
+                .get(&x)
+                .map_or(0.0, |count| f64::from(*count) / f64::from(num_samples));
+            let expected_probability = (1.0 - p).powf(f64::from(x)) * p;
+            println!("x = {x}, Observed Probability = {observed_probability}, Expected Probability = {expected_probability}");
+            assert!((observed_probability - expected_probability) <= 0.01);
```

Comment:
You should set your tolerance value and number of iterations as constants.

Reply:
okay